plugins/command-manager/src/main/java/com/wazuh/commandmanager/jobscheduler/CommandManagerJobRunner.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package com.wazuh.commandmanager.jobscheduler;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.jobscheduler.spi.JobExecutionContext;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.jobscheduler.spi.ScheduledJobRunner;
+import org.opensearch.threadpool.ThreadPool;
+
+import com.wazuh.commandmanager.CommandManagerPlugin;
+
+public class CommandManagerJobRunner implements ScheduledJobRunner {
+
+    private static final Logger log = LogManager.getLogger(CommandManagerJobRunner.class);
+    private static CommandManagerJobRunner INSTANCE;
+    private ThreadPool threadPool;
+    private ClusterService clusterService;
+
+    private Client client;
+    private final SearchJob searchJob = SearchJob.getInstance();
+
+    private CommandManagerJobRunner() {
+        // Singleton class, use getJobRunner method instead of constructor
+    }
+
+    public static CommandManagerJobRunner getInstance() {
+        log.info("Getting Job Runner Instance");
+        if (INSTANCE != null) {
+            return INSTANCE;
+        }
+        synchronized (CommandManagerJobRunner.class) {
+            if (INSTANCE != null) {
+                return INSTANCE;
+            }
+            INSTANCE = new CommandManagerJobRunner();
+            return INSTANCE;
```

Comment:
This can be simplified this like:
```
if (INSTANCE == null) {
        synchronized (CommandManagerJobRunner.class) {
            if (INSTANCE == null) {
                INSTANCE = new CommandManagerJobRunner();
            }
        }
    }
    return INSTANCE;
    ```